client_plugin/drivers/vfile/kvstore/etcdops/etcdops.go

```diff
@@ -289,20 +305,100 @@ func (e *EtcdKVS) joinEtcdCluster(leaderAddr string) error {
 	}
 
 	// start the routine for joining an etcd cluster
-	go etcdService(lines)
+	e.etcdStartService(lines)
 
 	// check if successfully joined the etcd cluster, then start the watcher
 	return e.checkLocalEtcd()
 }
 
-// etcdService function starts a routine of etcd
-func etcdService(cmd []string) {
-	_, err := exec.Command("/bin/etcd", cmd...).Output()
+// leaveEtcdCluster function is called when a manager is demoted
+func (e *EtcdKVS) leaveEtcdCluster() error {
+	nodeAddr := e.nodeAddr
```

Comment:
not an error, just curious - why not use e.nodeAddr where needed, why the extra vars ?

Reply:
I am trying to follow some rule to avoid multiple accesses to a parameter inside a struct...
But maybe it's not applicable here.
I can replace with using e.nodeAddr directly.